python/export_xcfun.cpp

```diff
@@ -0,0 +1,152 @@
+ */
+
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+
+#include "XCFun/xcfun.h"
+
+// Not ideal, I think
+#include "xcint.hpp"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace xcfun {
+PYBIND11_MODULE(_xcfun, m) {
+  m.doc() = "XCFun";
+
+  py::enum_<xc_mode>(m, "xc_mode")
+      .value("XC_MODE_UNSET", xc_mode::XC_MODE_UNSET)
+      .value("XC_PARTIAL_DERIVATIVES", xc_mode::XC_PARTIAL_DERIVATIVES)
+      .value("XC_POTENTIAL", xc_mode::XC_POTENTIAL)
+      .value("XC_CONTRACTED", xc_mode::XC_CONTRACTED)
+      .value("XC_NR_MODES", xc_mode::XC_NR_MODES)
+      .export_values();
+
+  py::enum_<xc_vars>(m, "xc_vars")
+      .value("XC_VARS_UNSET", xc_vars::XC_VARS_UNSET)
+      .value("XC_A", xc_vars::XC_A)
+      .value("XC_N", xc_vars::XC_N)
+      .value("XC_A_B", xc_vars::XC_A_B)
+      .value("XC_N_S", xc_vars::XC_N_S)
+      .value("XC_A_GAA", xc_vars::XC_A_GAA)
+      .value("XC_N_S_GNN_GNS_GSS", xc_vars::XC_N_S_GNN_GNS_GSS)
+      .value("XC_A_GAA_LAPA", xc_vars::XC_A_GAA_LAPA)
+      .value("XC_A_GAA_TAUA", xc_vars::XC_A_GAA_TAUA)
+      .value("XC_N_GNN_LAPN", xc_vars::XC_N_GNN_LAPN)
+      .value("XC_N_GNN_TAUN", xc_vars::XC_N_GNN_TAUN)
+      .value("XC_A_B_GAA_GAB_GBB_LAPA_LAPB", xc_vars::XC_A_B_GAA_GAB_GBB_LAPA_LAPB)
+      .value("XC_A_B_GAA_GAB_GBB_TAUA_TAUB", xc_vars::XC_A_B_GAA_GAB_GBB_TAUA_TAUB)
+      .value("XC_N_S_GNN_GNS_GSS_LAPN_LAPS", xc_vars::XC_N_S_GNN_GNS_GSS_LAPN_LAPS)
+      .value("XC_N_S_GNN_GNS_GSS_TAUN_TAUS", xc_vars::XC_N_S_GNN_GNS_GSS_TAUN_TAUS)
+      .value("XC_A_B_GAA_GAB_GBB_LAPA_LAPB_TAUA_TAUB",
+             xc_vars::XC_A_B_GAA_GAB_GBB_LAPA_LAPB_TAUA_TAUB)
+      .value("XC_A_B_GAA_GAB_GBB_LAPA_LAPB_TAUA_TAUB_JPAA_JPBB",
+             xc_vars::XC_A_B_GAA_GAB_GBB_LAPA_LAPB_TAUA_TAUB_JPAA_JPBB)
+      .value("XC_N_S_GNN_GNS_GSS_LAPN_LAPS_TAUN_TAUS",
+             xc_vars::XC_N_S_GNN_GNS_GSS_LAPN_LAPS_TAUN_TAUS)
+      .value("XC_A_AX_AY_AZ", xc_vars::XC_A_AX_AY_AZ)
+      .value("XC_A_B_AX_AY_AZ_BX_BY_BZ", xc_vars::XC_A_B_AX_AY_AZ_BX_BY_BZ)
+      .value("XC_N_NX_NY_NZ", xc_vars::XC_N_NX_NY_NZ)
+      .value("XC_N_S_NX_NY_NZ_SX_SY_SZ", xc_vars::XC_N_S_NX_NY_NZ_SX_SY_SZ)
+      .value("XC_A_AX_AY_AZ_TAUA", xc_vars::XC_A_AX_AY_AZ_TAUA)
+      .value("XC_A_B_AX_AY_AZ_BX_BY_BZ_TAUA_TAUB",
+             xc_vars::XC_A_B_AX_AY_AZ_BX_BY_BZ_TAUA_TAUB)
+      .value("XC_N_NX_NY_NZ_TAUN", xc_vars::XC_N_NX_NY_NZ_TAUN)
+      .value("XC_N_S_NX_NY_NZ_SX_SY_SZ_TAUN_TAUS",
+             xc_vars::XC_N_S_NX_NY_NZ_SX_SY_SZ_TAUN_TAUS)
+      .value("XC_A_2ND_TAYLOR", xc_vars::XC_A_2ND_TAYLOR)
+      .value("XC_A_B_2ND_TAYLOR", xc_vars::XC_A_B_2ND_TAYLOR)
+      .value("XC_N_2ND_TAYLOR", xc_vars::XC_N_2ND_TAYLOR)
+      .value("XC_N_S_2ND_TAYLOR", xc_vars::XC_N_S_2ND_TAYLOR)
+      .value("XC_NR_VARS", xc_vars::XC_NR_VARS)
+      .export_values();
+
+  m.def("xcfun_version", &xcfun_version, "XCFun version");
+  m.def("xcfun_splash", &xcfun_splash, "XCFun splash screen");
+
```

Comment:
Could you please add 
` m.def("xcfun_test", &xcfun_test, "XCFun self test");`

We are calling xcfun_test from the PyADF test suite to make sure xcfun is working correctly.

Reply:
Thanks for the review. Good catch! pybind11 won't generate binding code unless you explicitly define the function/class to be bound.